### PR TITLE
Smart pipe balance - mapped pipes change

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
All the mapped pipes set up their connectable direction to the ones already connected and disable all the others. This will help in preventing issues where under floor pipes connect in unwanted ways. 

The pipes are still able to connect to the constructed ones but only from the already made direction. 

You might require a bit more work to modify the basic setups but this will tremendously boost the new and old players experience with pipes

## Why It's Good For The Game
Many players (both old and new) have issues with invisible pipes that are under floors and connect in unwanted ways, while I believe that using simple tools like t-ray goggles and device will help you prevent wrong pipes from being placed in the wrong place it seems that the majority of people are still not using them. 
I hope thil will prevent unwanted delamination, waste to distro connections and other issues 

## Changelog
:cl:
qol: mapped pipes setup their available connection to only their current directions, preventing unwanted connections from the usual incompetent engineer/atmos tech
/:cl:

### CODE REVIEW

**Comment by GuillaumePrata** [comment]
So wait, all roundstart pipes will have their directions locked?
Wasn't the point of smart pipes to make expanding systems easier to build/rebuild? This basically forces us to go back to how things were before smart pipes.
I only see accidents like these happening because grey pipes exist *and* are the default.
I had a few accidents because I tried to change a pipe color, it didn't register because of TGUI or lag and I slapped a grey pipe in the wrong place.

**Comment by CRITAWAKETS** [comment]
There ARE a few issues with how piping is organized, though thats mostly mapping being stupid, like the SM just changing colors willy-nilly.

**Comment by Mothblocks** [comment]
> Wasn't the point of smart pipes to make expanding systems easier to build/rebuild? 

Yes, though reading the PR description a few times I can still hardly tell what this does. 

What precisely does this change, and does it prevent someone from just slapping a bunch of pipes where there used to be ones in order to connect it back to distro/waste?

**Comment by Ghilker** [comment]
I'll make a couple of images to explain the change better later on, but this doesn't prevent you from ease of rebuild, what it does is set the mapped pipes available connections to the initial ones. A straight pipe will connect to the extremes only and not on the sides, or a manifold won't connect to the not open side

**Comment by Ghilker** [comment]
let's say the vertical pipe is a mapped one, even if you put another blue pipe near it, it won't automatically connect because the mapped pipe's connections have been setted on roundstart. 
image
But i can replace that pipe with a new one and you have all the connections you need
image
this works even if you place a pipe in the place of a broken pipe

**Comment by Mothblocks** [comment]
In order to connect it, what do you have to do?

You say with a new pipe, does that mean unwrenching it and placing a new one? Won't that next one not connect? Won't that be a problem if distro is in such a way that can't be unwrenched?

Is there an intuitive way outside of unwrenching?

**Comment by Unit2E** [comment]
Honestly? Not a fan whatsoever, mostly because of what Mothblocks has already sort of alluded to. Changing around pipes around is one of the least forgiving and least fun aspects of Atmospherics as a whole, and resolving a part of this unfun aspect is one of the strongest points of smart pipes as a whole. 

Removing incorrect pipes is still an extremely punishing prospect, as you lose the gas in this pipe (sometimes for better, but often for worse), you get flung back, and gas spills on the tile of what you're unwrenching. These things can obviously be bad, if a pipe contains hot gas then that pipe can not ever be removed entirely safely without a good amount of extra effort. Anyway, this "problem" (arguable) is not entirely resolved with smart pipes, but the power to add to pipe nets without experiencing these things is what makes smart pipes good.

Currently, you can add filters to distro simply by adding pipes as a branch and placing filters on that branch, or you can speed up pressuring an area by adding injectors to distro. Same for waste, you can add things to it without needing to unwrench anything and go through the pains of gas leaks, fires, gas losses, being flung around, etcetera. 

All this change does is give into the people that haven't gotten used to smart pipes and demand reverts by stripping a feature off smart pipes that causes issues for people not used to it. Just to be clear to Mothblocks, if this change is implemented, you would have to unwrench and replace any and all mapped pipes that you intend to connect to any extension of your own. In the example Ghilker provided, they have unwrenched the pipe, deleted it, and replaced it, to then allow it to connect with the user-laid pipes to the left of it.

TL;DR: This change removes one of smart pipe's main advantages (for distro and waste) purely to satisfy people that don't know how colours and layers work yet and forces people to once again interact with the (in my opinion) unfun as hell unwrenching-a-pipe mechanics

**Comment by LemonInTheDark** [comment]
I don't like this, for the above reasons and partially because I don't like the idea of inconsistencies between laid pipes and roundstart pipes. See my long arguments with tim about layering.
Why am I agreeing with unit2 that guy used to be an admin, god.

I know the problem you're trying to fix, and I think the first step to it is forcing people to engage with color more actively, I think mainly by changing the default color option from grey. Hopefully this prevents people from just using it as a crutch, and avoids the "oops" memes we've seen in the past.

That and I've heard some fucked behavior around layer adapters and smart pipes? could just be hallucinating that tho.

**Comment by Mothblocks** [comment]
I'm not convinced this is a good idea, and looks like it would just be confusing.
